Zend/zend_objects.c

```diff
@@ -276,6 +276,55 @@ ZEND_API void ZEND_FASTCALL zend_objects_clone_members(zend_object *new_object,
 	}
 }
 
+ZEND_API zend_object *zend_objects_clone_obj_with(zend_object *old_object, const zend_class_entry *scope, const HashTable *properties)
+{
+	zend_object *new_object = old_object->handlers->clone_obj(old_object);
+
+	if (EXPECTED(!EG(exception)) && zend_hash_num_elements(properties) > 0) {
```

Comment:
`zend_hash_num_elements(properties) > 0` technically no longer needed. I would personally ASSERT and indicate that `clone_obj` should be called in this case.